posthog/taxonomy/property_definition_api.py

```diff
@@ -374,7 +393,23 @@ class Meta:
             "is_seen_on_filtered_events",
         )
 
-    def update(self, property_definition: PropertyDefinition, validated_data):
+    def validate(self, data):
+        validated_data = super().validate(data)
+
+        if "hidden" in validated_data and "verified" in validated_data:
+            if validated_data["hidden"] and validated_data["verified"]:
+                raise serializers.ValidationError("A property cannot be both hidden and verified")
+
+        return validated_data
+
+    def update(self, property_definition: PropertyDefinition, validated_data: dict):
+        # If setting hidden=True, ensure verified becomes false
+        if validated_data.get("hidden", False):
+            validated_data["verified"] = False
```

Comment:
I don't fully understand why there are two view sets for updating prop defs, but maybe we should update the logic here to be the same, where you also clear the verified at and verified by.